flink-connectors/flink-connector-base/src/test/java/org/apache/flink/connector/base/sink/writer/AsyncSinkWriterTest.java

```diff
@@ -862,6 +867,40 @@ private void writeTwoElementsAndInterleaveTheNextTwoElements(
                 "Executor Service stuck at termination, not terminated after 500ms!");
     }
 
+    @Test
+    public void ifTheNumberOfUncompletedInFlightRequestsIsTooManyThenBlockInFlushMethod()
+            throws Exception {
+        CountDownLatch blockedWriteLatch = new CountDownLatch(1);
+        CountDownLatch delayedStartLatch = new CountDownLatch(1);
+        AsyncSinkWriterImpl sink =
+                new AsyncSinkReleaseAndBlockWriterImpl(
+                        sinkInitContextAnyThreadMailbox,
+                        1,
+                        blockedWriteLatch,
+                        delayedStartLatch,
+                        false);
+
+        Thread t =
+                new Thread(
+                        () -> {
+                            try {
+                                sink.writeWithNonMailboxThread("1");
+                                sink.writeWithNonMailboxThread("2");
+                                sink.writeWithNonMailboxThread("3");
+                            } catch (IOException | InterruptedException e) {
+                                e.printStackTrace();
+                            }
+                        });
+        t.start();
+
+        delayedStartLatch.await();
```

Comment:
I think a doc string would help here to understand what the test waits for

Reply:
Definitely. Let me add this.